fix: support hashing with the new COMMITTED state

robinbryce opened the pull request.

Commit:
fix: support hashing with the new COMMITTED state

honourfish commented on simplehash/options.go:
```diff
@@ -57,8 +57,8 @@ func WithPublicFromPermissioned() HashOption {
 }
 
-func WithAsConfirmed() HashOption {
+func WithAsCommitted() HashOption {
```
Do we still need AsConfirmed?

What happens when a customer wants to update their tier and they get the ability to confirm their events?

Commit:
feat: v3 schema for forestrie

This new schema is required in order to ommit confirmation_status from
the hash. For simplehash assets, this field was crucial. For forestrie
it is only informative. And crucially, it changes after the event is
first added to the log STORED -> COMMITTED -> CONFIRMED -> UNEQUIVOCAL

We took the opportunity to prune other un-necessary fields:

* asset_identity is redundant and will soon be confusing.
* from is not relevant for forestrie (it was the custodial wallet
  address for khipu)

Note that idtimestamp is NOT added. The details of the leaf hash
construction are governed by forestrie and only change if the merkle log
format version is incremented. And the database can't store the log
values in native format anyway.

Additionally,

ci: enable code quality and unit tests on ci
